testdata/p4_16_samples/pna-mux-dismantle.p4

```diff
@@ -0,0 +1,216 @@
+
+struct main_metadata_t {
+    bit<1> rng_result1;
+}
+
+// User-defined struct containing all of those headers parsed in the
+// main parser.
+struct headers_t {
+    ethernet_t ethernet;
+    ipv4_t ipv4;
+    tcp_t tcp;
+}
+
+control PreControlImpl(
+    in    headers_t  hdr,
+    inout main_metadata_t meta,
+    in    pna_pre_input_metadata_t  istd,
+    inout pna_pre_output_metadata_t ostd)
+{
+    apply {
+    }
+}
+
+parser MainParserImpl(
+    packet_in pkt,
+    out   headers_t       hdr,
+    inout main_metadata_t main_meta,
+    in    pna_main_parser_input_metadata_t istd)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition select(hdr.ethernet.etherType) {
+            0x0800: parse_ipv4;
+            default: accept;
+        }
+    }
+    state parse_ipv4 {
+        pkt.extract(hdr.ipv4);
+        transition accept;
+    }
+}
+
+bit<1> dummy_func(inout headers_t hdr,   bit<16> min1,
+        bit<16> max1,inout main_metadata_t user_meta){
+    return  (bit<1>) ((min1 <= hdr.tcp.srcPort) && (hdr.tcp.srcPort <= max1));
+}
+
+action do_range_checks_0 (inout headers_t hdr,inout main_metadata_t user_meta,
+        bit<16> min1,
+        bit<16> max1) {
+    user_meta.rng_result1  = dummy_func(hdr,min1, max1, user_meta );
+}
+
+control MainControlImpl(
+    inout headers_t       hdr,           // from main parser
+    inout main_metadata_t user_meta,     // from main parser, to "next block"
+    in    pna_main_input_metadata_t  istd,
+    inout pna_main_output_metadata_t ostd)
+{
+    action next_hop(PortId_t vport) {
+        send_to_port(vport);
+    }
+    action add_on_miss_action() {
+        bit<32> tmp = 0;
+        add_entry(action_name="next_hop", action_params = tmp);
+    }
+
+    action do_range_checks_1 (
+        bit<16> min1,
+        bit<16> max1)
+    {
+        user_meta.rng_result1 =  (bit<1>) ((min1 <= hdr.tcp.srcPort) && (hdr.tcp.srcPort <= max1));
+    }
+
+    table ipv4_da {
+        key = {
+            hdr.ipv4.dstAddr: exact;
+
+        }
+        actions = {
+            @tableonly next_hop;
+            @defaultonly add_on_miss_action;
+        }
+        add_on_miss = true;
+        const default_action = add_on_miss_action;
+    }
+    action next_hop2(PortId_t vport, bit<32> newAddr) {
+        send_to_port(vport);
+        hdr.ipv4.srcAddr = newAddr;
+    }
+    action add_on_miss_action2() {
+        add_entry(action_name="next_hop", action_params = {32w0, 32w1234});
+    }
+    table ipv4_da2 {
+        key = {
+            hdr.ipv4.dstAddr: exact;
+        }
+        actions = {
+            @tableonly next_hop2;
+            @defaultonly add_on_miss_action2;
+            do_range_checks_1;
+            do_range_checks_0(hdr,user_meta);
+        }
+        add_on_miss = true;
+        const default_action = add_on_miss_action2;
+    }
+    apply {
+        user_meta.rng_result1  = (bit<1>) ((100 <= hdr.tcp.srcPort) && (hdr.tcp.srcPort <= 200));
```

Comment:
Is it the candidate ternary expression? not using "?", ":"?

Reply:
It becomes candidate after LowerExpressions pass because LowerExpressions itself introduce ternary, Writing directly in p4 does not reproduce the problem.

Reply:
Ok.